crates/starknet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -76,10 +78,24 @@ impl JsonRpcHandler {
     /// starknet_getClassHashAt
     pub(crate) async fn get_class_hash_at(
         &self,
-        _block_id: BlockId,
-        _contract_address: ContractAddressHex,
+        block_id: BlockId,
+        contract_address: ContractAddressHex,
     ) -> RpcResult<ClassHashHex> {
-        Err(error::ApiError::ContractNotFound)
+        let parsed_address = contract_address.0.try_into().map_err(|_| {
+            error::ApiError::RpcError(RpcError::invalid_params(format!(
+                "Invalid contract_address: {:?}",
+                contract_address.0
+            )))
+        })?;
+
+        let starknet = self.api.starknet.read().await;
+        let state = starknet
```

Comment:
We can move this logic in `Starknet` class itself. Then it can be reused by other calls. Let's just propagate call from `JsonRpcHandler` to `Starknet`. And handle Result in `JsonRpcHandler`

Reply:
This way we also won't need getter `.get_state_reader_at`